### PR TITLE
e2e test for bring your own keys with trustroot.

### PR DESCRIPTION
**Pull request by vaikas**
Signed-off-by: Ville Aikas <vaikas@chainguard.dev>

#### Summary
Add an e2e test for bringing your own keys with TrustRoot.
Remove the LogID from transparencylogs. It is automatically created for both Rekor as well as CTLog, so there's no need to expose it to the users. Note, it's still there in the intermediate representation because we want to reconcile it and use it during validation.

### CODE REVIEW

**Comment by vaikas** [comment]
FWIW, this is now seemingly blocked on:
https://github.com/sigstore/cosign/issues/2537

I'm seeing this in the verify logs, which would make sense :)
```
  error: failed to create job: admission webhook "policy.sigstore.dev" denied the request: validation failed: failed policy: image-policy-keyless-with-trustroot-with-attestations: spec.template.spec.containers[0].image
  registry.local:5000/policy-controller/demo@sha256:7ba401315d3ca3b154f03e71d69deb9eff90eb683a01cd9a4084531eb8624a41 attestation keyless validation failed for authority attestation for registry.local:5000/policy-controller/demo@sha256:7ba401315d3ca3b154f03e71d69deb9eff90eb683a01cd9a4084531eb8624a41: no matching attestations:
  error verifying bundle: verifying bundle: rekor log public key not found for payload

```

File: test/testdata/trustroot/e2e/bring-your-own-keys.yaml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+spec:
+  sigstoreKeys:
+    certificateAuthorities:
+    - subject:
+        organization: fulcio-organization
+        commonName: fulcio-common-name
+      uri: https://fulcio.fulcio-system.svc
+      certChain: |-
+        FULCIO_CERT_CHAIN
+    tLogs:
```
Comment (hectorj2f) [inline test/testdata/trustroot/e2e/bring-your-own-keys.yaml:28]: I know we added this before. What about using `tlogs` and `ctlogs` as the name here ?

File: go.mod [hunk 2]
```diff
@@ -273,4 +274,6 @@ replace github.com/golang/glog => github.com/jdolitsky/glog v0.0.0-2022072917223
 
 // Bring in the latest cosign so we can pass various keys in to verification
 // functions
-replace github.com/sigstore/cosign => github.com/sigstore/cosign v1.13.2-0.20221209171251-6cb723fc299c
+replace github.com/sigstore/cosign/v2 => github.com/vaikas/cosign/v2 v2.0.0-20221215003938-48e6404789cc
+
+// replace github.com/sigstore/cosign => github.com/sigstore/cosign v1.13.2-0.20221209171251-6cb723fc299c
```
Comment (hectorj2f) [inline go.mod:279]: can we remove this line ?

Comment (vaikas) [reply go.mod:279]: I'll wait until upstream fix goes in, I just wanted to test against my branch to make sure this PR is ready to go as soon as the upstream merges, I'll just rebase to that.

File: test/testdata/trustroot/e2e/bring-your-own-keys.yaml (hunk 1 above)
Comment (vaikas) [reply test/testdata/trustroot/e2e/bring-your-own-keys.yaml:28]: If we do the rename (which I don't have strong feelings about), I'd rather do those renames as part of a PR that only does that, you ok with that?

File: .github/workflows/kind-cluster-image-policy-trustroot.yaml [hunk 3]
```diff
@@ -89,11 +89,12 @@ jobs:
 
         # Wait for the webhook to come up and become Ready
         kubectl rollout status --timeout 5m --namespace cosign-system deployments/webhook
+        echo "TUF_ROOT_FILE=./root.json" >> $GITHUB_ENV
 
-    - name: Run Cluster Image Policy Tests that only tests keys, no keyless
-      timeout-minutes: 15
+    - name: Run Cluster Image Policy Tests that tests with bring your own keys
+      timeout-minutes: 5
```
Comment (hectorj2f) [inline .github/workflows/kind-cluster-image-policy-trustroot.yaml:95]: nice

**Review by hectorj2f — APPROVED** [review]
lgtm

**Review by hectorj2f — APPROVED** [review]
lgtm

**Pull request merged**